tests/WP_SQLite_Translator_Tests.php

```diff
@@ -421,6 +421,37 @@ public function testShowCreateTablePreservesDoubleUnderscoreKeyNames() {
 		);
 	}
 
+	public function testShowCreateTableLimitsKeyLengths() {
+		$this->assertQuery(
+			"CREATE TABLE _tmp__table (
+					`id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+					`order_id` bigint(20) unsigned DEFAULT NULL,
+					`meta_key` varchar(255) DEFAULT NULL,
+					`meta_value` text DEFAULT NULL,
+					PRIMARY KEY (`id`),
+					KEY `meta_key_value` (`meta_key`(100),`meta_value`(82)),
+					KEY `order_id_meta_key_meta_value` (`order_id`,`meta_key`(100),`meta_value`(82))
+				);"
+		);
+
+		$this->assertQuery(
+			'SHOW CREATE TABLE _tmp__table;'
+		);
+		$results = $this->engine->get_query_results();
+		$this->assertEquals(
+			'CREATE TABLE `_tmp__table` (
+	`id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+	`order_id` bigint(20) unsigned DEFAULT NULL,
+	`meta_key` varchar(255) DEFAULT NULL,
+	`meta_value` text DEFAULT NULL,
+	PRIMARY KEY (`id`),
+	KEY `order_id_meta_key_meta_value` (`order_id`,`meta_key`(100),`meta_value`(82)),
+	KEY `meta_key_value` (`meta_key`(100),`meta_value`(82))
```

Comment:
@JanJakes, currently, it skips all lengths.

As we don't store the lengths, do you have any idea how to approach that in a hotfix? Assume 100 for any varchar or text fields?

Reply:
@wojtekn Using something like 100 sounds like a reasonable hotfix. Maybe we could go for 191, as [WordPress seems to use that value](https://github.com/WordPress/wordpress-develop/blob/9104190b6d36c2959f652e0b65bf5bb7af42ffcc/src/wp-admin/includes/schema.php#L53).

Reply:
When I was fixing these CREATE_TABLE statements manually to attempt using it for import in Jetpack, I used `191` for length as well and that fixed the issue for me and I was able to work with these statements. This was an example:

```
DROP TABLE IF EXISTS `wp_wc_orders_meta`;
CREATE TABLE `wp_wc_orders_meta` (
	`id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
	`order_id` bigint(20) unsigned DEFAULT NULL,
	`meta_key` varchar(255) DEFAULT NULL,
	`meta_value` text DEFAULT NULL,
	PRIMARY KEY (`id`),
	KEY `order_id_meta_key_meta_value` (`order_id`, `meta_key`, `meta_value`(191)),
	KEY `meta_key_value` (`meta_key`, `meta_value`(191))
);
```

Reply:
> @wojtekn Using something like 100 sounds like a reasonable hotfix. Maybe we could go for 191, as [WordPress seems to use that value](https://github.com/WordPress/wordpress-develop/blob/9104190b6d36c2959f652e0b65bf5bb7af42ffcc/src/wp-admin/includes/schema.php#L53).

Thanks for sharing @JanJakes . It seems they calculated 191 assuming the limit of 767 bytes and `utf8mb4`, which uses 4 bytes per character. It works for the index with one field, but if we used such a value for an index that has more fields, it wouldn't work.

Currently, at least on WoA sites, the limit is 3072 bytes, and the collation used is `latin1_swedish_ci`, which uses 1 byte per character.  I played with the queries shown in the issue description, and it allowed me to add an index when I capped the text field length to 2809 characters. With one more, it failed:

```
	KEY `order_id_meta_key_meta_value` (`order_id`, `meta_key`, `meta_value`(2810))
```

However, if collation were set to `utf8mb4`, I would need to set it to 511 instead. If the limit for the index was 767 bytes, the `meta_key` alone would exceed it.

I propose going with an arbitrary value of 100 to keep things simple. With that one, for any varchar and text field, for given settings, it could support:
- `utf8mb4` and limit of 767 bytes: one text/varchar field plus a few integers
- `utf8mb4` and limit of 3072 bytes: seven text/varchar fields plus a few integers
- `latin1_swedish_ci` and limit of 767 bytes: seven text/varchar fields plus a few integers
- `latin1_swedish_ci` and limit of 3072 bytes: 30 text/varchar fields plus a few integers

If we dropped it to 80, in the first, worst-case scenario, it would support two text/varchar fields and a few integers. Thoughts?

Reply:
@wojtekn

> It works for the index with one field, but if we used such a value for an index that has more fields, it wouldn't work.

Ah, right!

> I propose going with an arbitrary value of 100 to keep things simple.
> If we dropped it to 80, in the first, worst-case scenario, it would support two text/varchar fields and a few integers.

That sounds reasonable to me. I think 80 is fine as well. Unless some string values are expected to share very long prefixes, I actually don't see much need to index that much. That said, WP tries to make use of the 191 limit, so pushing it to something like 80 or 100 might be the way to go.

On the other hand, if we expect some plugins to index 3 string fields (which I'm sure can happen), then we may even go with something like 50. After all, these numbers are the maximums of what can be indexed, and I would expect the actual values to be much shorter in most cases.